examples/lua/main.c

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright (C) 2018 FU Berlin
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     examples
+ * @{
+ *
+ * @file
+ * @brief       Basic lua example application
+ *
+ * @author      Daniel Petry <daniel.petry@fu-berlin.de>
+ *
+ * @}
+ */
+
+#include <stdio.h>
+#include <stdlib.h>
+#include "lauxlib.h"
+#include "lualib.h"
+
+#include "main.lua.h"
+
+int lua_run_script (const char *buffer, size_t buffer_len ){
+
+    lua_State *L = luaL_newstate();
+
+    if (L == NULL) {
+        puts("cannot create state: not enough memory");
+        return EXIT_FAILURE;
```

Comment:
@danpetry are this return codes LUA specific?

Reply:
No, they're defined in stdlib.h

Reply:
I would prefer here `errno` (e.g ENOBUFS)

Reply:
I've used ENOMEM for this one; EINTR for failure of the protected call to the lua engine, to run the script; and `return 0;` for success in line with most other examples I looked at.

Reply:
fine ;)